### ISSUE TITLE
nightqa fails in some early data when TILEID isn't defined in data header

### ISSUE THREAD

**Issue by akremin** [desihub/desispec#2356]
When running nightqa in Kibo we've hit an error when running on data from 2020 and early 2021.

Example:
`/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20201214.log`

```
Traceback (most recent call last):
  File "/global/common/software/desi/perlmutter/desiconda/20240425-2.2.0/code/desispec/0.66.2/bin/desi_night_qa", line 210, in <module>
    main()
  File "/global/common/software/desi/perlmutter/desiconda/20240425-2.2.0/code/desispec/0.66.2/bin/desi_night_qa", line 141, in main
    expids, tileids, surveys = get_surveys_night_expids(args.night)
  File "/global/common/software/desi/perlmutter/desiconda/20240425-2.2.0/code/desispec/0.66.2/lib/python3.10/site-packages/desispec/night_qa.py", line 136, in get_surveys_night_expids
    tileids.append(hdr["TILEID"])
  File "/global/common/software/desi/perlmutter/desiconda/20240425-2.2.0/conda/lib/python3.10/site-packages/fitsio/header.py", line 354, in __getitem__
    raise KeyError("unknown record: %s" % item)
KeyError: 'unknown record: TILEID'
```

**Comment by akremin** [comment]
While this is being fixed there is also a warning that appears since it can't find the SURVEY either:
`WARNING:night_qa.py:133:get_surveys_night_expids: SURVEY could not be identified for /dvs_ro/cfs/cdirs/desi/spectro/data/20201214/00067797/desi-00067797.fits.fz; setting to 'unknown'`

The KeyError and warning happened on 7 nights in Kibo:
```
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20201214.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20201215.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20210114.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20210218.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20210223.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20210331.log
/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/logs/nightqa-kibo-20210401.log
```

One of the scripts that is crashing due to the error:
`/global/cfs/cdirs/desi/spectro/redux/kibo/run/scripts/tsnr_nightqa/scripts/nightqa-kibo-20201214.slurm`

The actual command submitted that leads to the error:
`desi_night_qa --nproc=64 --recompute -p kibo -n 20201214  -o /global/cfs/cdirs/desi/spectro/redux/kibo/nightqa/20201214`

**Comment by araichoor** [comment]
about the error:

I ve checked for those night specifically: the issue is that there are some `SCIENCE` exposure with no fiberassign file.
so I guess we just need to discard those exposures; that s a simple fix -- and actually a shortcoming of the code...
I just need to handle the case where no fiberassign file is found here:
https://github.com/desihub/desispec/blob/33d11ac38df5c7e9b18cefe3d28b1cd1278b6dac/py/desispec/night_qa.py#L126-L131

for the record, here are these "problematic" exposures for the nights listed above:
```
# NIGHT EXPID EXPTIME
20201214	00067797	2.0
20201214	00067798	2.0
20201214	00067799	2.0
20201215	00067801	2.0
20201215	00067802	2.0
20210114	00072382	10.0206
20210114	00072383	900.0525
20210114	00072384	900.052
20210114	00072385	900.0642
20210114	00072386	900.0578
20210114	00072387	900.0514
20210114	00072388	900.065
20210114	00072391	10.0209
20210114	00072392	600.0411
20210114	00072393	600.0427
20210114	00072394	600.0321
20210114	00072395	600.03
20210114	00072396	600.0299
20210114	00072397	600.0432
20210114	00072398	600.0382
20210114	00072399	600.0315
20210114	00072400	600.0493
20210114	00072401	600.0363
20210114	00072402	600.0349
20210114	00072403	600.0238
20210114	00072404	600.0352
20210114	00072405	600.0402
20210114	00072406	600.0413
20210218	00076671	10.0203
20210218	00076779	37.5934
20210218	00076786	180.0555
20210218	00076788	180.0554
20210218	00076794	180.0538
20210223	00077639	30.0594
20210223	00077640	30.0604
20210223	00077641	30.0601
20210223	00077661	30.0601
20210223	00077662	30.0594
20210223	00077663	30.0599
20210223	00077687	30.0596
20210223	00077688	30.0599
20210223	00077689	30.0602
20210331	00082808	5.0092
20210401	00083013	11.5816
20210401	00083014	9.6684
```

**Comment by araichoor** [comment]
about the warning:
I ve to check, but I think it s ok; I think it s only used in `create_petalnz_pdf()`, to decide which tiles to consider for that diagnosis; as said here https://github.com/desihub/desispec/issues/2357#issuecomment-2333290020, only `sv1`, `sv2`, `sv3`, `main` surveys are considered; so `unknown` will just -- rightly -- be ignored there.

**Comment by akremin** [comment]
I confirmed that all of the exposures in your list are set to be ignored by the pipeline, so they will not appear in Kibo or any future prod. It would be safe to skip them, or just set their tileid to -99 (which is what I did in the pipeline exposure_tables).

**Comment by akremin** [comment]
Resolved with Pull Request #2361
